File: src/main/java/org/iesvdm/proyecto/model/Aula.java
package org.iesvdm.proyecto.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = false)
@Entity
public class Aula {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private long id;
    private String grupo;
    @Column(nullable = false)
    private String año;
    @ManyToMany
    private Set<Profesor> profesores;
    @ManyToOne(fetch = FetchType.EAGER,optional = false)
    private Profesor propietario;
    @ManyToMany
    private Set<Estudiante> estudiantes;
    @ManyToOne(optional = false)
    private Clase clase;
    @OneToMany(mappedBy = "aula",cascade = CascadeType.ALL,orphanRemoval = true)
    private Set<Tema> temas;
    private boolean eliminado;
    public String getRoute(){
        return this.clase.getRoute()+" "+this.grupo+" "+this.año;
    }
}

File: src/main/frontend/app-proyecto/src/app/interfaces/aula.ts
import {Option} from "./option";
import {Tema} from "./tema";

export interface AulaForm {
  id:number,
  clase:Option,
  grupo:string,
  anio:string,
}
export interface Aula {
  id:number,
  clase:string,
  grupo:string,
  anio:string,
  temas:Tema[],
  propietario:number
}
